test/Quest1155.t.sol

```diff
@@ -0,0 +1,306 @@
+        );
+        assertEq(
+            SampleERC1155(sampleERC1155).balanceOf(participant, TOKEN_ID),
+            1,
+            "participant should have received the reward"
+        );
+    }
+
+    // todo add fuzz test
+
+    function test_RevertIf_singleClaim_NotQuestFactory() public {
+        SampleERC1155(sampleERC1155).mintSingle(address(quest), 1, 100000);
+        vm.deal(address(quest), 100000000);
+        vm.prank(questFactoryMock);
+        quest.queue();
+
+        vm.expectRevert(abi.encodeWithSelector(NotQuestFactory.selector));
+        quest.singleClaim(participant);
+    }
+
+    function test_RevertIf_singleClaim_NotStarted() public {
+        SampleERC1155(sampleERC1155).mintSingle(address(quest), 1, 100000);
+        vm.deal(address(quest), 100000000);
+        vm.prank(questFactoryMock);
+        quest.queue();
+
+        vm.warp(START_TIME - 1);
+        vm.prank(questFactoryMock);
+        vm.expectRevert(abi.encodeWithSelector(NotStarted.selector));
+        quest.singleClaim(participant);
+    }
+
+    function test_RevertIf_singleClaim_whenNotPaused() public {
+        SampleERC1155(sampleERC1155).mintSingle(address(quest), 1, 100000);
+        vm.deal(address(quest), 100000000);
+        vm.prank(questFactoryMock);
+        quest.queue();
+        vm.startPrank(questFactoryMock);
+        quest.pause();
+        vm.warp(START_TIME);
+        vm.expectRevert("Pausable: paused");
+        quest.singleClaim(participant);
+        vm.stopPrank();
+    }
+
+    // /*//////////////////////////////////////////////////////////////
+    //                   WITHDRAWREMAININGTOKENS
+    // //////////////////////////////////////////////////////////////*/
+
+    function test_withdrawRemainingTokens() public {
+        SampleERC1155(sampleERC1155).mintSingle(address(quest), 1, 100000);
+        vm.deal(address(quest), 100000000);
+        vm.prank(questFactoryMock);
+        quest.transferOwnership(owner);
+        vm.prank(owner);
+        quest.queue();
+
+        uint256 ownerOGBalance = owner.balance;
+        vm.warp(END_TIME + 1);
+        vm.prank(protocolFeeRecipient);
+
+        quest.withdrawRemainingTokens();
+        assertEq(
+            owner.balance,
+            ownerOGBalance + 100000000,
+            "owner should have received remaining ETH"
+        );
+        assertEq(
+            SampleERC1155(sampleERC1155).balanceOf(owner, TOKEN_ID),
+            100000,
+            "owner should have received remaining ERC1155"
+        );
+    }
+
+    // todo add fuzz test
+
+    function test_RevertIf_withdrawRemainingToken_NotEnded() public {
+        SampleERC1155(sampleERC1155).mintSingle(address(quest), 1, 100000);
+        vm.deal(address(quest), 100000000);
+        vm.prank(questFactoryMock);
+        quest.transferOwnership(owner);
+        vm.prank(owner);
+        quest.queue();
+
+        vm.expectRevert(abi.encodeWithSelector(NotEnded.selector));
```

Comment:
Might be good to standardize the error messages between this and `Quest.t.sol` should we capture this in a separate issue? Out of scope for this PR.